### ISSUE TITLE
teleport causing OBS to hang on exit (macOS)

### ISSUE THREAD

**Issue by yamyamyamyamyam** [fzwoch/obs-teleport#33]
Hi again, 

I've noticed that on macOS, Teleport is frequently causing my OBS to hang and consume resources on exit, regardless of whether any Teleport sources are added to any scenes. I took a spindump, which you can find here:

https://gist.github.com/yamyamyamyamyam/ab8a125b9f9e57b46da32e598c3b611f

My reproduction steps (does not occur every time, but pretty often):

1. Open OBS
2. Open settings
3. Close settings
4. Close OBS

Let me know if there's anything else I can do to help debug this. Thanks!

**Comment by fzwoch** [comment]
Is 2. and 3. the OBS settings, or the Teleport Output preference dialog?

The debugger throws weird things when trying to open the preference dialog. I can't make any sense out of it though. It seems to break when the callback for the menu is supposed to be called, but it does not even reach there.

Does this happen with the x86_64 version on either native or running under Rosetta the same?

At the moment I have no idea why this is happening.

**Comment by yamyamyamyamyam** [comment]
2 and 3 are the OBS settings, which is why I found it odd; I was not directly interacting with Teleport to trigger the hang.

I'm using a nightly build with native arm support as well as Apple's new General Capture code. I just tried with the old official 27.2.4 stable x86_64 build, and I can't trigger the hang there. I tried with a nightly x86_64 build, and the issue does occur there.

So there might be some bisecting in order to figure out exactly what's causing the regression. Possibly you'd just want to wait until there is a settled release candidate or something.
